### PR TITLE
doc: improved grammar, spelling, and punctuation

### PR DESCRIPTION
**Pull request by SteLeo1602**
Finished Pull Request #14730 which worked on improving grammar, spelling, and punctuation.

### CODE REVIEW

**Comment by SteLeo1602** [comment]
GitHub, generate website

**Comment by SteLeo1602** [comment]
@nrmancuso This is a reminder to review my changes in this pull request. It's been over a week and if I remember correctly, I should notify you after that period of time.

**Comment by nrmancuso** [comment]
> @nrmancuso This is a reminder to review my changes in this pull request. It's been over a week and if I remember correctly, I should notify you after that period of time.

Sorry for the delay, and thank you for the reminder. We have been busy with GSOC, I will get to your PR in the next couple of days. I appreciate your patience, @SteLeo1602

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 1]
```diff
@@ -619,12 +621,12 @@ JAVADOC -> JAVADOC [0:0]
...
-        that support it. A custom check needs to extend <tt>AbstractJavadocCheck</tt> for having
-        this functionality readily available. Do note that a check which has this property set to
-        true will log violation only for the first not-tight HTML tag found. To allow a check to
-        skip processing of javadocs with non-tight HTML, method
+        Checks can also be configured to log violations upon encountering non-tight HTML tags.
+        The <tt>violateExecutionOnNonTightHtml</tt> property can be used for this purpose in the
+        checks that support it. A custom check needs to extend <tt>AbstractJavadocCheck</tt> to
+        have this functionality readily available. Do note that a check which has this property set
+        to true, will log violations only for the first not-tight HTML tag found. To allow a check
+        to skip processing of javadocs with non-tight HTML, method
```
Comment (nrmancuso) [inline src/xdocs/writingjavadocchecks.xml.vm:629]:
```suggestion
        to skip processing of javadocs with non-tight HTML, the <tt>acceptJavadocWithNonTightHtml</tt> method in 
```

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 2]
```diff
@@ -743,7 +745,7 @@ Checkstyle ends with 5 errors.
 
     <section name="Checkstyle SDK GUI">
       <p>
-       Checkstyle GUI allows to show javadoc tree in java files. To run in use
+       Checkstyle GUI allows to show javadoc trees in java files. To run it, use
```
Comment (nrmancuso) [inline src/xdocs/writingjavadocchecks.xml.vm:748]:
```suggestion
       The Checkstyle GUI provides us with a way to show javadoc trees in java files. To run it, use
```

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 3]
```diff
@@ -774,15 +776,15 @@ Checkstyle ends with 5 errors.
...
       <p>
-        Java checks controlled by method <a href="writingchecks.html#Understanding_token_sets">
+        Java checks are controlled by methods <a href="writingchecks.html#Understanding_token_sets">
           setTokens(), getDefaultTokens(), getAccessibleTokens(), getRequiredTokens(). </a>
-        JavaDoc checks use the same model plus extra 4 methods for Javadoc tokens.
-        As Java AST and Javadoc AST are not binded.
-        It is highly recommended for Javadoc checks do not use customization of java tokens and
-        expect to be executed only on javadoc tokens.
+        JavaDoc checks use the same model plus 4 additional methods for Javadoc tokens.
+        As Java AST and Javadoc AST are not binded,
```
Comment (nrmancuso) [inline src/xdocs/writingjavadocchecks.xml.vm:782]:
```suggestion
        As Java AST and Javadoc AST are not bound,
```

**Review by nrmancuso — CHANGES_REQUESTED** [review]
First pass, thanks for removing the part about lazy maintainers:

**Comment by nrmancuso** [comment]
@SteLeo1602 this part of our documentation is wild, please feel free to remove unnecessary adjectives and anything that seems to wordy or not valuable.

File: src/xdocs/beginning_development.xml [hunk 4]
```diff
@@ -43,7 +43,7 @@
 
     <section name="Starting Development">
       <p>
-        Here you can find instructions of importing and debugging the project for IDEs:<br/>
+        Here you can find instructions on importing and debugging the project for IDEs:<br/>
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/beginning_development.xml]: Done

File: src/xdocs/config_system_properties.xml [hunk 5]
```diff
@@ -28,18 +28,18 @@
...
         <p>
-          The following is an example of include xml files content in xml by ENTITY feature
-          to let keep common part of configs in single file and composite main config
-          by few smaller parts.
+          The following is an example of including the contents of other xml files by using the
+          ENTITY feature to keep common parts of configs in a single file and then creating
+          composite configs from smaller parts.
           <br/>
-          Imagine we want to define for test sources a bit different requirements
+          Imagine we want to define slightly different requirements for test sources
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/config_system_properties.xml]: Done

File: src/xdocs/contributing.xml [hunk 6]
```diff
@@ -55,20 +55,21 @@
     <section name="Report an issue">
 
       <p>
-        All functional changes in project should have registered issue and wide explanation in it.
-        Issue number has to be referenced in git commit message, see format below.
+        All functional changes in the project should have a registered issue and detailed
+        explanation in it. The Issue number has to be referenced in the git commit message,
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/contributing.xml]: Done

File: src/xdocs/contributing.xml [hunk 7]
```diff
@@ -87,13 +88,7 @@
...
 
       <p>
-        The developer team of checkstyle is really a lazy bunch of
-        people.  We try to avoid work as best as we can, but most of all
-        we try to avoid working on bugs that are reported by end users.
-      </p>
-
-      <p>
-        To that end, we use a set of development tools that ensure that
+        We use a set of development tools that ensure that
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/contributing.xml]: Done

File: src/xdocs/contributing.xml [hunk 8]
```diff
@@ -55,20 +55,21 @@
...
+        To report an issue please follow our best practices page - <a href="report_issue.html">
         How to report an issue?</a> (<a href="report_issue.html#How_to_report_a_bug.3F">
         How to report a bug?</a>)
       </p>
     </section>
 
     <section name="Build">
       <p>
-        Project is following general maven layout and phases for build.
+        The project is following a general maven layout and phases for its build.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/contributing.xml]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 9]
```diff
@@ -533,14 +535,14 @@ JAVADOC -> JAVADOC [0:0]
         <table style="table-layout: fixed;">
           <tr>
             <td>
-              1) Unclosed paragraph HTML tag. As you see in the tree, content of the paragraph tag
-              is not nested to this tag.
-              That is because HTML tags are not closed by pair tag &lt;/p&gt;, and Checkstyle
+              1) Unclosed paragraph HTML tag. As you see in the tree, the content of the paragraph
+              tag is not nested to this tag.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 10]
```diff
@@ -533,14 +535,14 @@ JAVADOC -> JAVADOC [0:0]
...
+              That is because HTML tags are not closed by pair tags &lt;/p&gt;, and Checkstyle
               requires <a href="#Tight-HTML_rules">Tight-HTML</a> code to predictably parse
               Javadoc comments.
             </td>
             <td>
-              2) Here is correct version with open and closed HTML tags.
+              2) Here is a correct version with open and closed HTML tags.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm (hunk 1 above)
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm (hunk 2 above)
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm (hunk 3 above)
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/contributing.xml [hunk 11]
```diff
@@ -107,7 +102,7 @@
         <br/>
         <code>mvn test jacoco:restore-instrumented-classes jacoco:report@default-report</code>.
         <br/>
-        Check results on report target/site/jacoco/index.html in project home folder.
+        Check the results of the report in target/site/jacoco/index.html in project home folder.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/contributing.xml]: Done

File: src/xdocs/index.xml.vm [hunk 12]
```diff
@@ -72,10 +72,11 @@
...
       <p>
         Current site contains documentation only for latest, released version.
-        We only support this latest version. We highly recommend read latest documentation first.
+        We only support this latest version. We highly recommend reading the latest documentation
+        first.
         <br/>
         However, you might need documentation for previous releases because you are still using
-        an older version and latest version describe some features that does not work for you.
+        an older version and the latest version describes some features that don't work for you.
         <br/>
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/index.xml.vm]: Done

File: src/xdocs/index.xml.vm [hunk 13]
```diff
@@ -72,10 +72,11 @@
     <section name="Previous Version Documentation">
       <p>
         Current site contains documentation only for latest, released version.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/index.xml.vm]: Done

File: src/xdocs/index.xml.vm [hunk 14]
```diff
@@ -167,10 +168,10 @@
         </table>
       </div>
       <p>
-        Checkstyle currently is confirmed to be build by all JDKs from 11 through 17.
+        Checkstyle currently is confirmed to be buildable by all JDKs from 11 through 17.
         Most recent JDKs may be supported. Please
         <a href="https://checkstyle.org/report_issue.html">report an issue</a>
-        if there is any problems with most recent JDKs.
+        if there are any problems with most recent JDKs.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/index.xml.vm]: Done

File: src/xdocs/contributing.xml [hunk 15]
```diff
@@ -127,8 +122,8 @@
       </p>
 
       <p>
-        The last step of verification that all works fine please do testing of your
-        functionality on any open-source project (Spring, Hibernate, ....).
+        For the last step of verification, to ensure that everything works fine, please test
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/contributing.xml]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 16]
```diff
@@ -99,15 +99,16 @@ public class MyClass {
...
-      All tags are copied as is to result javadoc html pages by Sun/Oracle javadoc tool.
-      All bad formatting is responsibility of user and web-browser. To validate Checkstyle to parse
-      input to predictable structure - Abstract Syntax Tree(AST). It is very difficult to parse
-      free style format, so input text need to follow some format, so limitation appears.
+      Javadoc by specification could contain any HTML tags that lets users generate the content
+      they need. All tags are copied as is to result in javadoc html pages by the Sun/Oracle
+      javadoc tool. All bad formatting is the responsibility of user and web-browser. To validate
+      Checkstyle it needs to parse input into a predictable structure - Abstract Syntax Tree(AST).
+      Since it is very difficult to parse a free style format, input text needs to follow some
+      format, therefore limitations apply.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 17]
```diff
@@ -99,15 +99,16 @@ public class MyClass {
...
       </p>
       <p>
       The comment should be written in <a href="#Tight-HTML_rules">Tight-HTML</a>
-          to build nested AST Tree that most Checks expect.
+          to build nested ASTs that most Checks expect.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 18]
```diff
@@ -127,14 +128,14 @@ public class MyClass {
...
         <a href="https://www.w3.org/TR/html5/tabular-data.html#the-tr-element">TR</a>), so,
-        Checkstyle won't show error about missing end tag, however,
-        it leads to broken Tight-HTML structure and as a result
-        leads to not-nested content of the HTML tags in Abstract Syntax Tree of the Javadoc comment.
+        Checkstyle won't show an error about missing end tags, however,
+        it leads to broken Tight-HTML structure and as a result leads to non-nested content
+        of the HTML tags in the Abstract Syntax Tree of the Javadoc comment.
         <br/>
-        In other words, if HTML tags are not closed Javadoc grammar cannot determine content
+        In other words, if HTML tags are not closed Javadoc, grammar cannot determine content
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 19]
```diff
@@ -127,14 +128,14 @@ public class MyClass {
...
         of these tags, so structure of the parse tree will not be nested like it is while using
-        <a href="#Tight-HTML_rules">Tight-HTML</a> code. It is done just to not fail on every
-        Javadoc comment, because there are tons of using unclosed tags, etc.
+        <a href="#Tight-HTML_rules">Tight-HTML</a> code. This is done just to not fail on every
+        Javadoc comment, because a ton of unclosed tags are used, etc.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 20]
```diff
@@ -157,7 +158,7 @@ public class MyClass {
     <section name="How to create Javadoc Check">
       <p>
         Principle of writing Javadoc Checks is similar to writing regular Checks. You just extend
-        another abstract class and use another token types.
+        another abstract class and use another set of token types.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 21]
```diff
@@ -194,10 +195,10 @@ public class MyClass {
...
       <p>
-        Java grammar parses java file base on Java language specifications. So, there are
+        Java grammar parses java file based on Java language specifications. So, there are
         singleline comments and multiline/block comments in it.
-        Java compiler doesn't know about Javadoc because it is just a multiline comment.
-        To parse multiline comment as a Javadoc comment, checkstyle has special Parser
+        The Java compiler doesn't know about Javadoc because it is just a multiline comment.
+        To parse a multiline comment as a Javadoc comment, checkstyle has a special Parser
         that is based on ANTLR Javadoc grammar. So, it's supposed to process block comments
         that start with Javadoc Identifier and parse them to Abstract Syntax Tree (AST).
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 22]
```diff
@@ -211,11 +212,12 @@ public class MyClass {
...
-        are no whitespace/newline nodes. In Javadoc comment every whitespace matters, and Javadoc
-        Checks need all those whitespaces and newline nodes to verify format and content of the
-        Javadoc comment. Because of that Javadoc grammar includes all whitespaces, newlines to
-        the parse tree
+        The main Java grammar skips any whitespaces and newlines, so there are no
+        whitespace/newline nodes in Java Abstract Syntax Tree.
+        there are no whitespace/newline nodes. In Javadoc comments every whitespace matters, and
+        Javadoc Checks need all those whitespaces and newline nodes to verify format and content of
+        the Javadoc comment. Because of that, the Javadoc grammar includes all whitespaces, and
+        newlines to the parse tree
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 23]
```diff
@@ -286,15 +288,15 @@ CLASS_DEF -> CLASS_DEF [5:0]
     `--RCURLY -> } [7:0]
       ]]></source>
       <p>
-        As you see very small java file transforms to a huge Abstract Syntax Tree, because that is
-        the most detailed tree including all components of the java file: classes, methods,
+        As you see, a very small java file transforms into a huge Abstract Syntax Tree, because that
+        is the most detailed tree including all components of the java file: classes, methods,
         comments, etc.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 24]
```diff
@@ -340,20 +342,20 @@ JAVADOC -> JAVADOC [0:0]
 
     <section name="Access Java AST from Javadoc Check">
       <p>
-        As you already know Javadoc parse tree is a result of parsing block comment. There is a
-        method to get the original block comment from Javadoc Check.
-        You may need this block comment to check its position or something else in java
+        As you already know the Javadoc parse tree is a result of parsing the block comment.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 25]
```diff
@@ -340,20 +342,20 @@ JAVADOC -> JAVADOC [0:0]
...
+        There is a method to get the original block comment from a Javadoc Check.
+        You may need this block comment to check its position or something else in a java
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 26]
```diff
@@ -340,20 +342,20 @@ JAVADOC -> JAVADOC [0:0]
...
         <a href="apidocs/com/puppycrawl/tools/checkstyle/api/DetailAST.html">DetailAST</a> tree.
       </p>
       <p>
-        For example, to write a JavadocCheck that verifies @param tags in Javadoc comment of
-        a method definition, you also need all method's parameter names. To get method
-        definition AST you should access java
+        For example, to write a JavadocCheck that verifies @param tags in the Javadoc comment of
+        a method definition, you also need all of the method's parameter names. To get a method
+        definition AST you should access the java
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 27]
```diff
@@ -396,7 +398,7 @@ class MyCheck extends AbstractJavadocCheck {
         </a>.
       </p>
       <p>
-        HTML4 is picked just to have a list of elements whose end tag is optional(omittable) and a
+        HTML4 was made to just have a list of elements whose end tag is optional(omittable) and a
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 28]
```diff
@@ -415,22 +417,22 @@ class MyCheck extends AbstractJavadocCheck {
...
         To make Checkstyle support HTML5 tags whose end tag is optional (omittable) and HTML5 void
-        elements we should update Javadoc Parser because each element that breaks
-        <a href="#Tight-HTML_rules">Tight-HTML rules</a> have to be defined in Javadoc grammar.
-        In the future, we should update Javadoc grammar if those tag lists extend (new tags,
+        elements, we should update the Javadoc Parser because each element that breaks
+        <a href="#Tight-HTML_rules">Tight-HTML rules</a> has to be defined in Javadoc grammar.
+        In the future, we should update the Javadoc grammar if those tag lists extend (new tags,
         new HTML standard, etc.).
         (We already have an <a href='https://github.com/checkstyle/checkstyle/issues/3332'>issue
         on updating Javadoc grammar to HTML5</a>)
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 29]
```diff
@@ -415,22 +417,22 @@ class MyCheck extends AbstractJavadocCheck {
...
       </p>
 
       <p>
-        If Checkstyle meets unknown tag (for example HTML5 tag)
-        it doesn't fail and parses this tag as
+        If Checkstyle meets an unknown tag (for example HTML5 tag)
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 30]
```diff
@@ -415,22 +417,22 @@ class MyCheck extends AbstractJavadocCheck {
...
+        it doesn't fail and parses this tag as a
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 31]
```diff
@@ -415,22 +417,22 @@ class MyCheck extends AbstractJavadocCheck {
...
         <a href="apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html#HTML_TAG">
             HTML_TAG</a>
           Javadoc token type.
-        Just follow <a href="#Tight-HTML_rules">Tight-HTML rules</a> to make
-          Checkstyle javadoc parser make nested AST, even though tags are unknown.
+        Just follow <a href="#Tight-HTML_rules">Tight-HTML rules</a> to make the
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 32]
```diff
@@ -470,8 +472,8 @@ JAVADOC -> JAVADOC [0:0]
       </p>
 
       <p>
-        Here is what you get if unknown tag doesn't have matching end tag (for example, HTML5 tag
-        &lt;audio&gt;): <br/>
+        Here is what you get if the unknown tag doesn't have a matching end tag (for example, HTML5
+        tag &lt;audio&gt;): <br/>
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 33]
```diff
@@ -483,9 +485,9 @@ JAVADOC -> JAVADOC [0:0]
         </source>
       </div>
       <p>
-        As you see Javadoc parser prints error and doesn't build AST if unknown HTML tag doesn't
-        have matching end tag. If that a case please create an issue against Checkstyle to
-        upgrade parser.
+        As you see, the Javadoc parser prints an error and doesn't build the AST if an unknown
+        HTML tag doesn't have a matching end tag. If that's the case, please create an issue for
+        Checkstyle to upgrade the parser.
```
[the remark this answers is not in the document]
Comment (SteLeo1602) [reply src/xdocs/writingjavadocchecks.xml.vm]: Done

**Review by nrmancuso — APPROVED** [review]
Big improvement, thank you very much @SteLeo1602 !

**Comment by romani** [comment]
Github, generate website

**Review by romani — APPROVED** [review]

File: src/xdocs/contributing.xml [hunk 34]
```diff
@@ -127,12 +122,11 @@
...
-        The last step of verification that all works fine please do testing of your
-        functionality on any open-source project (Spring, Hibernate, ....).
-        Here is how to do it by
-        <a href=
-          "https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#checkstyle-tester">
-        our testing tool</a>. Generated report is required to be shared in Pull Request.
+        We require regression testing for most functional changes, especially for check modules.
+        See our <a href=
+          "https://github.com/checkstyle/contribution/tree/master/
+          checkstyle-tester#checkstyle-tester">
```
Comment (rnveach) [inline src/xdocs/contributing.xml:128]: 404 error

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/xdocs/writingjavadocchecks.xml.vm [hunk 35]
```diff
@@ -414,23 +402,18 @@ class MyCheck extends AbstractJavadocCheck {
...
       <p>
-        To make Checkstyle support HTML5 tags whose end tag is optional (omittable) and HTML5 void
-        elements we should update Javadoc Parser because each element that breaks
-        <a href="#Tight-HTML_rules">Tight-HTML rules</a> have to be defined in Javadoc grammar.
-        In the future, we should update Javadoc grammar if those tag lists extend (new tags,
-        new HTML standard, etc.).
-        (We already have an <a href='https://github.com/checkstyle/checkstyle/issues/3332'>issue
-        on updating Javadoc grammar to HTML5</a>)
+        Checkstyle does not support HTML5 tags with optional end tags. See
+        <a href='https://github.com/checkstyle/checkstyle/issues/3332'/> for details.
```
Comment (rnveach) [inline src/xdocs/writingjavadocchecks.xml.vm:406]: 1) Switch to double quotes
2) Why is this self closing? This means this link has no clickable text.

**Review by rnveach — CHANGES_REQUESTED** [review]

**Comment by romani** [comment]
Oh, sorry, wrong PR....

**Comment by romani** [comment]
@SteLeo1602 , please address review items in new PR.

@rnveach , please do more review on this PR , I will fix your points.
Sorry for merging, I thought I am on different PR.

**Comment by romani** [comment]
@rnveach, I fixed your concerns - https://github.com/checkstyle/checkstyle/pull/15323 please review

**Pull request merged**